test/index.js

```diff
@@ -26,739 +26,615 @@ describe('Token', function () {
-        expect(res.statusCode).to.equal(401);
-        done();
-      });
+      const res = await newServer.inject(request);
+      expect(res.result.message).to.equal(jwtErrorPrefix + 'jwt subject invalid. expected: ' + subject);
+      expect(res.statusCode).to.equal(401);
     });
 
-    it('fails if token subject is invalid', function (done) {
-  
+    it('fails if token subject is invalid', async function () {
       var request = { method: 'POST', url: '/token', headers: { authorization: tokenHeader('john', {subject:'https://invalid.subject'}) } };
 
-      server.inject(request, function (res) {
-        expect(res.result.message).to.equal(jwtErrorPrefix + 'jwt subject invalid. expected: ' + subject);
-        expect(res.statusCode).to.equal(401);
-        done();
-      });
+      const res = await newServer.inject(request);
+      expect(res.result.message).to.equal(jwtErrorPrefix + 'jwt subject invalid. expected: ' + subject);
+      expect(res.statusCode).to.equal(401);
     });
 
-    it('works if token subject is valid', function (done) {
-  
+    it('works if token subject is valid', async function () {
       var request = { method: 'POST', url: '/token', headers: { authorization: tokenHeader('john', {subject: subject}) } };
 
-      server.inject(request, function (res) {
-        expect(res.result).to.exist;
-        expect(res.statusCode).to.equal(200);
-        done();
-      });
+      const res = await newServer.inject(request);
+      expect(res.result).to.exist;
+      expect(res.statusCode).to.equal(200);
     });
 
   });
 
 });
 
 describe('Strategy', function(){
-    
-  it('should fail if strategy is initialized without options', function (done) {
-    var server = new Hapi.Server({ debug: false  });
-    server.connection();
-    server.register(require('../'), function (err) {
-      expect(err).to.not.exist;
-      try {
-        server.auth.strategy('default', 'jwt', 'required');
-        done('Should have failed')
-      }
-      catch(err){
-        expect(err).to.exist;
-        expect(err.message).to.equal('"jwt auth strategy options" must be an object');
-        done();
-      }
-    });
-  });
 
-  it('should fail if strategy is initialized with a string as options', function (done) {
-    var server = new Hapi.Server({ debug: false  });
-    server.connection();
-    server.register(require('../'), function (err) {
-      expect(err).to.not.exist;
-      try {
-        server.auth.strategy('default', 'jwt', 'required', 'wrong options type');
-        done('Should have failed')
-      }
-      catch(err){
-        expect(err).to.exist;
-        expect(err.message).to.equal('"jwt auth strategy options" must be an object');
-        done();
-      }
-    });
+  it('should fail if strategy is initialized without options', async function () {
+    var server = new Hapi.Server({ debug: false });
+    await server.register(require('../'))
+    try {
+      server.auth.strategy('default', 'jwt', null);
```

Comment:
The test won't fail if this doesn't throw. You need to use `expect.fail` or so after this line.